app/services/crawlee_bridge.py

```
@@ -0,0 +1,292 @@
+            
+            return {
+                'saved_count': saved_count,
+                'duplicate_count': duplicate_count,
+                'error_count': error_count,
+                'job_ids': job_ids
+            }
+            
+        except Exception as e:
+            db.rollback()
+            logger.error(f"Database save failed: {e}")
+            raise CrawleeIntegrationError(f"Failed to save jobs to database: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)